File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.Queue;
import java.util.Arrays;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> answer = new ArrayList<>();
        
        Queue<Integer> progressQueue = new LinkedList<>();
        Queue<Integer> speedQueue = new LinkedList<>();
        
        for (int progress : progresses) progressQueue.add(progress);
        for (int speed : speeds) speedQueue.add(speed);
        
        int cnt = 0;
        int day = 0;
        
        while (!progressQueue.isEmpty()) {
            if (progressQueue.peek() + speedQueue.peek() * day >= 100) {
                cnt ++;
                progressQueue.remove();
                speedQueue.remove();
            } else {
                if (cnt > 0) {
                    answer.add(cnt);
                    cnt = 0;
                }
                day++;
            }
        }
        
        answer.add(cnt);
        
        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    answer = []
    
    day = 0
    cnt = 0
    
    pq = deque(progresses)
    sq = deque(speeds)
    
    while pq:
        if pq[0] + sq[0] * day >= 100:
            cnt += 1
            pq.popleft()
            sq.popleft()
        else:
            if cnt > 0:
                answer.append(cnt)
                cnt = 0
            day += 1
    
    answer.append(cnt)
    
    return answer